### ISSUE TITLE
False-positive ImpureMethodCall when calling method on an object that created inside of a pure method

### ISSUE THREAD

**Issue by fluffycondor** [vimeo/psalm#8413]
https://psalm.dev/r/0310c9c472

If I understand correctly, this code shouldn't yield an error. It blocks psalm/psalm-plugin-doctrine#125 now, so I'd like to know can I safely suppress it.

**Comment by AndrolGenhald** [comment]
Yeah, this can be safely suppressed, duplicate of #5615. As weirdan mentioned there, it seems cloning fixes this, so we need to make it so that creating a new object works the same way as cloning.

If you trick Psalm by creating and then cloning it works fine: https://psalm.dev/r/c75ed63c79

I think that issue will probably have to be fixed before merging that PR though, as I understand it the PR will cause everyone that uses `Criteria::orderBy` to have the false positive that they have to suppress, rather than just being a suppression in the plugin itself.
